tests/Unit/EmogrifierTest.php

```diff
@@ -2015,6 +2015,54 @@ public function secondNonImportantStyleNotOverwritesFirstImportantOne()
         );
     }
 
+    /**
+     * @test
+     */
+    public function emogrifyAppliesLaterOverridingStyleAfterStyleAfterOverriddenStyle()
+    {
+        $this->setSubjectBoilerplateHtml();
+        $this->subject->setCss('p { margin-top: 1px; } p { margin: 2px; } p { margin-top: 3px; }');
```

Comment:
These should be two separate tests: One that the (first) margin overrides the margin-top, and one that the (other) margin-top gets applies after the margin.

Reply:
Ok, I've added a separate test, which can be updated if/when elimination of individual properties overridden by shorthand properties is supported.